lib/client/doc.js

```diff
@@ -383,7 +383,8 @@ Doc.prototype._resubscribe = function() {
 
 // Request the current document snapshot or ops that bring us up to date
 Doc.prototype.fetch = function(callback) {
-  if (this.connection.canSend) {
+  // If there is an inflightFetch, prevent to trigger another one
+  if (this.connection.canSend && !this.inflightFetch.length) {
```

Comment:
I think you're right. In fact, we should not have many situations where we have to debounce a lot of fetches. The only case I see is where you trigger a fetch and you receive another op from the server before the answer of the fetch from the sever. This will (if the version is still lower than the current one) trigger another fetch. I agree to remove the debounce modification I did.